### PR TITLE
feat: continue-activity event is extended to support webpageURL property

### PR DESCRIPTION
**Pull request by 1akshat1**
#### Description of Change

This PR aims to extend the existing `continue-activity` event params to resolve issue https://github.com/electron/electron/issues/29926.
Primarily a new `details` object is added as a parameter to the event, and `webpageURL` is added inside that details object if the userActivity contains `webpageURL`. 
 
#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes:  Extended `continue-activity` event API to support `webpageURL` property from `NSUserActivity`.

### CODE REVIEW

**Comment by 1akshat1** [comment]
@nornagon I tried testing the changes locally but I'm getting the details object as undefined in my app. I don't have experience with objective C so figured I'll request your input. To test I copied the `Electron.app` generated by `e build` to my App's electron node module folder. (Not sure if that would work or not). 

I'm testing the changes manually by invoking Apple's handoff mechanism.

Notes on Apple's handoff mechanism for universal linking: 
1. Context can be passed from the Apple's browser to a native App.
2. To achieve this, `com.apple.developer.associated-domains` entitlement must be placed on the server for the webpageURL we'd like universally link.
3. The web domain which is to be linked needs to be added to the app's `entitlement.plist`.
4. Once everything is in place and the packaged app is in Applications folder on Mac, macOs would be able to open the app when a user clicks the universal link rather than opening the link on browser.

More info: 
https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html
https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/Handoff/AdoptingHandoff/AdoptingHandoff.html#//apple_ref/doc/uid/TP40014338-CH2-SW10

**Comment by nornagon** [comment]
Are you using electron-builder / electron-forge or something like it to create the .app bundle for your app to place it in /Applications? I think that will ignore whatever Electron.app is in your node_modules folder and download Electron itself...

**Comment by 1akshat1** [comment]
> Are you using electron-builder / electron-forge or something like it to create the .app bundle for your app to place it in /Applications? I think that will ignore whatever Electron.app is in your node_modules folder and download Electron itself...

Yeah, that seems to be the case. I tried using electron-builder's property `electronDist` to point it to the custom build but it is throwing `corrupted Electron dist` error. Probably has to do with plist issue I think.

**Comment by 1akshat1** [comment]
@nornagon I was able to test this out using a boilerplate electron app and using `electron-builder`'s `electronDist` property to use the custom Electron build. It is working as expected. I used an existing Apple provisioning profile to sign the build and an existing apple-app-site-association file on the server to complete Apple universal linking setup. I'm not sure how we could achieve this in an automated manner though.

File: docs/api/app.md [hunk 1]
```diff
@@ -161,6 +161,8 @@ Returns:
   [`NSUserActivity.activityType`][activity-type].
 * `userInfo` unknown - Contains app-specific state stored by the activity on
   another device.
+* `details` Object
+  * `webpageURL` String - A string identifying the URL of the webpage accessed by the activity on another device.
```
Comment (nornagon) [inline docs/api/app.md:165]: It looks like this isn't always present, so let's make sure to mark it as such.
```suggestion
  * `webpageURL` String (optional) - A string identifying the URL of the webpage accessed by the activity on another device, if available.
```

**Review by nornagon — APPROVED** [review]
LGTM after docs update 👍 

I'm OK without an automated test for this.

**Comment by 1akshat1** [comment]
@nornagon Seems like the build is failing on linux.

**Review by jkleinsc — APPROVED** [review]

**Comment by jkleinsc** [comment]
API LGTM

**Pull request merged**